engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -849,14 +850,16 @@ - (void)viewDidAppear:(BOOL)animated {
     [self onUserSettingsChanged:nil];
     [self onAccessibilityStatusChanged:nil];
     BOOL stateIsActive = YES;
-#if APPLICATION_EXTENSION_API_ONLY
-    if (@available(iOS 13.0, *)) {
-      stateIsActive = self.flutterWindowSceneIfViewLoaded.activationState ==
-                      UISceneActivationStateForegroundActive;
+
+    UIApplication* flutterApplication = [FlutterSharedApplication uiApplication];
+    if (flutterApplication != nil) {
+      stateIsActive = flutterApplication.applicationState == UIApplicationStateActive;
+    } else {
+      if (@available(iOS 13.0, *)) {
+        stateIsActive = self.flutterWindowSceneIfViewLoaded.activationState ==
+                        UISceneActivationStateForegroundActive;
```

Comment:
Since these are enums, we should use `switch` statements for coverage checking in the lints in the unlikely case future states are ever added. Similar to the background check we do in `FlutterEngine.mm` above, this has the nice side effect of nudging us to extract this check to a method (in the internal API), which improves readability.